lib/index.js

```diff
@@ -623,19 +623,25 @@ Socket.prototype.send = function(msg, flags, cb) {
   return this;
 };
 
-
-Socket.prototype._flushRead = function () {
-  var message = this._zmq.readv(); // can throw
-  if (!message) {
-    return false;
-  }
-
-  // Handle received message immediately to prevent memory leak in driver
-  if (message.length === 1) {
+Socket.prototype._emitMessages = function (messages) {
```

Comment:
The name of this method is confusing, because it actually only emits one message. I think it should be `Socket.prototype._emitMessage = function (message)`